### PR TITLE
add remaining PropTypes on components

### PR DESCRIPTION
**Pull request by mikebond84**
I am not very confident about these, would like to review in class with everyone.

### CODE REVIEW

File: src/components/AddStack.js [hunk 1]
```diff
@@ -24,4 +25,8 @@ class AddNewStack extends Component {
   }
 }
 
+AddNewStack.PropTypes = {
```
Comment (cndreisbach) [inline src/components/AddStack.js]: `AddNewStack.PropTypes` should be `AddNewStack.propTypes`. In addition, this component doesn't really have props -- it only has state.

File: src/components/App.js [hunk 2]
```diff
@@ -87,4 +88,11 @@ const Guard = ({ redirectTo, condition, children }) => {
   }
 }
 
+App.propTypes = {
+  currentUser: PropTypes.shape({
```
Comment (cndreisbach) [inline src/components/App.js]: This also appears to be referring to state, not props. `App` should generally not have props.

File: src/components/BigFlashCard.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 
 const BigFlashCard = (props) => <div className='BigFlashCard'>{props.children}</div>
 
+BigFlashCard.PropTypes = {
+  children: PropTypes.object
```
Comment (cndreisbach) [inline src/components/BigFlashCard.js]: `children` are a special prop -- they are the components contained inside this component. We leave them out of PropTypes checking.

File: src/components/FlashCardMini.js [hunk 4]
```diff
@@ -10,4 +11,8 @@ const FlashCardMini = (props) =>
     </div>
   </div>
 
+FlashCardMini.PropTypes = {
```
Comment (cndreisbach) [inline src/components/FlashCardMini.js]: `PropTypes` -> `propTypes`

File: src/components/FlashCardMini.js [hunk 5]
```diff
@@ -10,4 +11,8 @@ const FlashCardMini = (props) =>
...
+  cardFront: PropTypes.object
```
Comment (cndreisbach) [inline src/components/FlashCardMini.js]: This should be a string, not an object

File: src/components/LoginForm.js [hunk 6]
```diff
@@ -68,4 +69,13 @@ class LoginForm extends React.Component {
   }
 }
 
+LoginForm.propTypes = {
```
Comment (cndreisbach) [inline src/components/LoginForm.js]: This component has one prop -- setCurrentUser, which is a function.

File: src/components/RegistrationForm.js [hunk 7]
```diff
@@ -78,4 +79,14 @@ class RegistrationForm extends React.Component {
   }
 }
 
+RegistrationForm.propTypes = {
```
Comment (cndreisbach) [inline src/components/RegistrationForm.js]: This component has one prop -- setCurrentUser, which is a function.

File: src/components/RunMode.js [hunk 8]
```diff
@@ -42,7 +42,7 @@ class RunMode extends React.Component {
   }
 }
 
-RunMode.propTypes = {
+RunMode.PropTypes = {
```
Comment (cndreisbach) [inline src/components/RunMode.js]: This change should be removed.

File: src/components/Spinner.js [hunk 9]
```diff
@@ -1,5 +1,9 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 
 const Spinner = ({ className }) => (<div className={`Spinner ${className}`} />)
 
+Spinner.PropTypes = {
```
Comment (cndreisbach) [inline src/components/Spinner.js]: `PropTypes` -> `propTypes`. In addition, this component takes one prop -- className, which is a string.

File: src/components/StackTitle.js [hunk 10]
```diff
@@ -36,4 +37,10 @@ class StackTitle extends Component {
   }
 }
 
+StackTitle.PropTypes = {
```
Comment (cndreisbach) [inline src/components/StackTitle.js]: `PropTypes` -> `propTypes`

File: src/components/StackTitle.js [hunk 11]
```diff
@@ -36,4 +37,10 @@ class StackTitle extends Component {
...
+  title: PropTypes.shape({
```
Comment (cndreisbach) [inline src/components/StackTitle.js]: `title` is a prop, but it should be a string

**Comment by mikebond84** [comment]
I'm going to close this and do another pull request if I can with changes.

**Pull request closed without merging**